src/utils.jl

```diff
@@ -195,12 +195,24 @@ function __get_bcresid_prototype(::StandardBVProblem, prob::BVProblem, u)
     return prototype, size(prototype)
 end
 
-@inline function __fill_like(v, x, args...)
+@inline function __similar(x, args...)
     y = similar(x, args...)
+    eltype(y) <: BigFloat && fill!(y, BigFloat(0))
```

Comment:
is there a reason to not always fill with zeros?

Reply:
zeros is better, just change them all. While NonlinearSolve is doing the similar thing like specialize on bigfloat and do `__similar` things in https://github.com/SciML/NonlinearSolve.jl/pull/438, maybe we can change to zeros everywhere in NonlinearSolve.jl as well?

Reply:
We should. https://github.com/SciML/SciMLStyle?tab=readme-ov-file#always-default-to-constructs-which-initialize-data

Reply:
Note this will also fix issues with non-fully defined functions. For example, if a user doesn't write to `du[i]`, we should still converge now, while before you'd get something random. So a few bugs should be fixed, some bugs avoided, etc. Avoiding uninitialized memory is just a good idea all around and NonlinearSolve needs to do this change too.